client/tests/integration_tests/add_domain.rs

```diff
@@ -36,3 +36,41 @@ fn client_add_domain_with_name_length_more_than_limit_should_not_commit_transact
 
     Ok(())
 }
+
+#[test]
+fn add_domain_with_logo() {
+    use std::str::FromStr;
+
+    use iroha_data_model::domain::IpfsPath;
+
+    // Invalid paths
+    assert!(matches!(
+        IpfsPath::from_str(""),
+        Err(err) if err.to_string() == "Expected root type, but nothing found"
+    ));
+    assert!(matches!(
+        IpfsPath::from_str("/ipld"),
+        Err(err) if err.to_string() == "Expected at least one content id"
+    ));
+    assert!(matches!(
+        IpfsPath::from_str("/ipfs/a"),
+        Err(err) if err.to_string() == "IPFS cid is too short"
+    ));
+    assert!(matches!(
+        IpfsPath::from_str("/ipfsssss/QmQqzMTavQgT4f4T5v6PWBp7XNKtoPmC9jvn12WPT3gkSE"),
+        Err(err) if err.to_string() == "Unexpected root type"
+    ));
+
+    // Valid paths
+    IpfsPath::from_str("QmQqzMTavQgT4f4T5v6PWBp7XNKtoPmC9jvn12WPT3gkSE").unwrap();
+    IpfsPath::from_str("/ipfs/QmQqzMTavQgT4f4T5v6PWBp7XNKtoPmC9jvn12WPT3gkSE").unwrap();
+    IpfsPath::from_str("/ipld/QmQqzMTavQgT4f4T5v6PWBp7XNKtoPmC9jvn12WPT3gkSE").unwrap();
+    IpfsPath::from_str("/ipns/QmSrPmbaUKA3ZodhzPWZnpFgcPMFWF4QsxXbkWfEptTBJd").unwrap();
+    IpfsPath::from_str("/ipfs/SomeFolder/SomeImage").unwrap();
+
+    let logo = IpfsPath::from_str("/ipfs/QmQqzMTavQgT4f4T5v6PWBp7XNKtoPmC9jvn12WPT3gkSE").unwrap();
```

Comment:
Everything above this line should go into a unit test in `data_model/stc/lib.rs`.